workspaces/global-floating-action-button/plugins/global-floating-action-button/src/types.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum Slot {
+  PAGE_END = 'page-end',
+  BOTTOM_CENTER = 'bottom-center',
+}
+
+export type FloatingActionButtonWithPositions = Record<
+  Slot | string,
+  FloatingActionButton[]
+>;
+
+export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
+
+export type FloatingActionButton = {
+  label: string;
+  showLabel?: boolean;
+  icon: string | React.ReactElement;
```

Comment:
As mentioned before, I recommend to make the icon optional:

```suggestion
  icon?: string | React.ReactElement;
```